File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {

    static double GIGASECOND = 1E9;
    LocalDateTime momentPlusGigaSec;

    public Gigasecond(LocalDate moment) {
        this(moment.atStartOfDay());
    }

    public Gigasecond(LocalDateTime moment) {
        this.momentPlusGigaSec = moment.plusSeconds((long) GIGASECOND);
    }

    public LocalDateTime getDateTime() {
        return momentPlusGigaSec;
    }
}

File: exercism-scala/gigasecond/src/main/scala/Gigasecond.scala
import java.time.{LocalDate, LocalDateTime}

object Gigasecond {

  val gigaseconds = 1E9

  def add(startDate: LocalDate): LocalDateTime = add(startDate.atStartOfDay())

  def add(startDateTime: LocalDateTime): LocalDateTime = startDateTime.plusSeconds(gigaseconds.toLong)
}
